src/commands/createFunction/openAPISteps/OpenAPICreateStep.ts

```diff
@@ -0,0 +1,89 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ProgressLocation, Uri, window } from "vscode";
+import { AzureWizardExecuteStep, DialogResponses, IActionContext } from 'vscode-azureextensionui';
+import { ProjectLanguage } from "../../../constants";
+import { ext } from "../../../extensionVariables";
+import { localize } from "../../../localize";
+import { cpUtils } from "../../../utils/cpUtils";
+import { nonNullProp } from '../../../utils/nonNull';
+import { openUrl } from '../../../utils/openUrl';
+import { IJavaProjectWizardContext } from '../../createNewProject/javaSteps/IJavaProjectWizardContext';
+import { IDotnetFunctionWizardContext } from '../dotnetSteps/IDotnetFunctionWizardContext';
+import { IFunctionWizardContext } from "../IFunctionWizardContext";
+
+export class OpenAPICreateStep extends AzureWizardExecuteStep<IFunctionWizardContext> {
+    public priority: number = 220;
+
+    public static async createStep(context: IActionContext): Promise<OpenAPICreateStep> {
+        await validateAutorestInstalled(context);
+        return new OpenAPICreateStep();
+    }
+
+    public async execute(wizardContext: IFunctionWizardContext & IJavaProjectWizardContext & IDotnetFunctionWizardContext): Promise<void> {
+        const uris: Uri[] = nonNullProp(wizardContext, 'openApiSpecificationFile');
+        const uri: Uri = uris[0];
+        const args: string[] = [];
+
+        args.push(`--input-file:${cpUtils.wrapArgInQuotes(uri.fsPath)}`);
+        args.push(`--version:3.0.6320`)
+
+        switch (wizardContext.language) {
+            case ProjectLanguage.TypeScript:
+                args.push('--azure-functions-typescript');
+                args.push('--no-namespace-folders:True');
+                break;
+            case ProjectLanguage.CSharp:
+                args.push(`--namespace:${nonNullProp(wizardContext, 'namespace')}`);
+                args.push('--azure-functions-csharp');
+                break;
+            case ProjectLanguage.Java:
+                args.push(`--namespace:${nonNullProp(wizardContext, 'javaPackageName')}`);
+                args.push('--azure-functions-java');
+                break;
+            case ProjectLanguage.Python:
+                args.push('--azure-functions-python');
+                args.push('--no-namespace-folders:True');
+                args.push('--no-async');
+                break;
+            default:
+                throw new Error(localize('notSupported', 'Not a supported language. We currently support C#, Java, Python, and Typescript'));
```

Comment:
You don't support JavaScript?

Reply:
Not yet. Once we do JS and PS, we'll be adding them as well.